libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs

```diff
@@ -359,22 +321,27 @@ private async Task ProcessSingleMessageAsync(FacebookMessage message, BotCallbac
         /// </summary>
         /// <param name="request">An Http request object.</param>
         /// <param name="response">An Http response object.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        private async Task<bool> VerifySignatureAsync(HttpRequest request, HttpResponse response)
+        private async Task<bool> VerifySignatureAsync(HttpRequest request, HttpResponse response, CancellationToken cancellationToken)
         {
             var expected = request.Headers["x-hub-signature"];
-            var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(this.options.AppSecret));
-
-            var bodyStream = new StreamReader(request.Body);
 
-            var calculated = "sha1=" + hmac.ComputeHash(Encoding.UTF8.GetBytes(bodyStream.ReadToEnd()));
+            string calculated = null;
+            using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(_options.AppSecret)))
```

Comment:
I remember that this encryption was used multiple times on different botkits, wouldnt be useful to create a static class to do this? So one could just do `calculated = EncryptHelper.ToSHA256(_options.AppSecret, request.Body)`

Reply:
maybe